functions/assignOfflineContact.ts

```diff
@@ -0,0 +1,141 @@
+import '@twilio-labs/serverless-runtime-types';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
+import {
+  responseWithCors,
+  bindResolve,
+  error400,
+  error500,
+  success,
+  send,
+} from '@tech-matters/serverless-helpers';
+// eslint-disable-next-line prettier/prettier
+import type { PromiseValue } from 'type-fest';
+
+const TokenValidator = require('twilio-flex-token-validator').functionValidator;
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  TWILIO_CHAT_TRANSFER_WORKFLOW_SID: string;
+};
+
+export type Body = {
+  targetSid?: string;
+  finalTaskAttributes: string;
+};
+
+// eslint-disable-next-line prettier/prettier
+type TaskInstance = PromiseValue<ReturnType<ReturnType<ReturnType<ReturnType<Context['getTwilioClient']>['taskrouter']['workspaces']>['tasks']>['fetch']>>;
+// eslint-disable-next-line prettier/prettier
+type WorkerInstance = PromiseValue<ReturnType<ReturnType<ReturnType<ReturnType<Context['getTwilioClient']>['taskrouter']['workspaces']>['workers']>['fetch']>>;
+
+const assignToAvailableWorker = async (
+  event: Body,
+  newTask: TaskInstance,
+) => {
+  const reservations = await newTask.reservations().list();
+  const reservation = reservations.find(r => r.workerSid === event.targetSid);
+
+  if (reservation) {
+    await reservation.update({ reservationStatus: 'accepted' });
+    await reservation.update({ reservationStatus: 'completed' });
+    return { type: 'success' } as const;
+  } 
+
+  await newTask.remove();
+  return {
+    type: 'error',
+    payload: { status: 500, message: 'Error: reservation for task not created.' },
```

Comment:
How will we handle this error when it is sent back to the frontend of the worker who entered the offline contact?

Reply:
The error 500 is handled in the frontend in the same way as the error in hrm, "backend error" pop-up alert. You can try it out by running the frontend with a bad serverless url.